### PR TITLE
10.0

### PR DESCRIPTION
**Pull request by lrvick**
Refactor and initial support for 10.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,67 +18,90 @@ contain := \
...
 	@$(contain) fetch
 
-tools: fetch image
+.PHONY: tools
+tools:
 	@$(contain) tools
 
+.PHONY: random
+random:
+	test -f $(PWD)/build/.rnd || head -c 1G </dev/urandom > $(PWD)/build/.rnd
```
Comment (drGrove) [inline Makefile]: `< /dev/urandom`

This was annoying me earlier

File: config.yml [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
 user: build
 platform:
-  ref: android-9.0.0_r45
+  ref: android-10.0.0_r2
   extra_remotes:
   - name: github
     fetch: https://github.com
   extra_projects:
-  - name: anestisb/android-prepare-vendor
+  - name: RattlesnakeOS/android-prepare-vendor
```
Comment (drGrove) [inline config.yml]: Why RattlesnakeOS?

File: scripts/build-kernel [hunk 3]
```diff
@@ -4,31 +4,20 @@
...
 #ln -vs \
 #	"${key_dir}/kernel.pem" \
 #	"${kernel_dir}/private/msm-google/certs/signing_key.pem"
-
+cd "${kernel_dir}"
 cat <<-EOF | bash
 	export OUT_DIR="${kernel_dir}/out"
 	export DIST_DIR="${kernel_dir}/dist"
-	bash "${kernel_dir}/build/build.sh" -j$(cores)
+	bash "${kernel_dir}/build/build.sh" -j "$cores"
```
Comment (drGrove) [inline scripts/build-kernel]: `"${cores}"`

File: scripts/fetch [hunk 4]
```diff
@@ -34,3 +43,34 @@ retry 3 repo sync \
...
+	--manifest-url "${manifest_dir}" \
+	--manifest-name "kernel-${kernel_name}.xml"
+repo sync -c --no-tags --no-clone-bundle --jobs "${cores}"
+
+repo forall -c 'git reset --hard ; git clean -fdx'
+
+# Chromium
+if [ "$build_chromium" == "true"  ]; then
+	[ ! -d "$chromium_dir/depot_tools" ] && \
+		git clone "${chromium_build_tools_url}" "$chromium_dir/depot_tools"
```
Comment (drGrove) [inline scripts/fetch]: `${chromium_dir}`

File: scripts/fetch [hunk 5]
```diff
@@ -34,3 +43,34 @@ retry 3 repo sync \
...
+	cd "$chromium_dir/depot_tools"
+	git fetch
+	git reset --hard "$chromium_depot_tools_ref"
```
Comment (drGrove) [inline scripts/fetch]: `${chromium_depot_tools_ref}`

File: scripts/fetch [hunk 6]
```diff
@@ -34,3 +43,34 @@ retry 3 repo sync \
...
+
+	export PATH="$PATH:$chromium_dir/depot_tools"
```
Comment (drGrove) [inline scripts/fetch]: This is not consistent, wrap env vars in `{}`

File: scripts/manifest [hunk 7]
```diff
@@ -166,15 +166,13 @@ if __name__ == "__main__":
...
     manifest_kernels=[]
     for device in config['devices'].keys():
-        kernel_name = config['devices'][device]['global']['kernel_name']
-        repo = config['devices'][device]['global']['kernel_repo']
-        ref = config['devices'][device]['global']['kernel_ref']
-        if kernel_name not in kernel_names and 'manifest' in repo:
+        kernel_name = config['devices'][device]['kernel_name']
+        repo = "kernel/manifest"
+        ref = config['devices'][device]['kernel_ref']
+        filename = '%s/kernel-%s.xml' % (path, kernel_name)
```
Comment (drGrove) [inline scripts/manifest]: Move to using .format() instead

**Review by drGrove — CHANGES_REQUESTED** [review]

File: config.yml (hunk 2 above)
Comment (lrvick) [reply config.yml]: their fork is the only one with 10 support atm.

File: scripts/build-kernel (hunk 3 above)
Comment (lrvick) [reply scripts/build-kernel]: ha. good catch. no wonder that was not working

**Comment by lrvick** [comment]
@drGrove Just tested on my taimen. Everything works.

**Review by drGrove — APPROVED** [review]

**Pull request merged**